mrp_subcontracting_partner_management/models/res_partner.py

```diff
@@ -6,258 +8,219 @@ class ResPartner(models.Model):
-        """Creating Subcontracting Location starts here"""
-
-        location_vals = {
-            "name": self._compose_entity_name(),
-            "usage": "internal",
-            "location_id": parent_location or False,
-            "company_id": company.id,
-            "active": True,
-        }
-        location_rec = self.subcontracted_created_location_id
-        if not location_rec:
-            location_rec = self.env["stock.location"].create(location_vals)
-        return location_rec
-
-    def _create_subcontracted_operation_type(self, warehouse, location):
+    def _create_subcontracted_operation_type(self, vals):
         """Creating Operation Type for Subcontracting"""
-        name = self._compose_entity_name()
-        operation_type_name = "{}: {}".format(name, " IN")
-        sequence_code = ""
-        for code in list(filter(None, operation_type_name.split(" "))):
-            sequence_code += code[0]
-        operation_type_rec = self.partner_picking_type_id
-        if not operation_type_rec:
-            operation_type_vals = {
-                "name": operation_type_name,
-                "code": "incoming",
-                "sequence_code": sequence_code,
-                "is_subcontractor": True,
-            }
-            if warehouse:
-                operation_type_vals.update({"warehouse_id": warehouse.id})
-            if location:
-                operation_type_vals.update({"default_location_dest_id": location.id})
-            operation_type_rec = self.env["stock.picking.type"].create(
-                operation_type_vals
-            )
-        return operation_type_rec
-
-    def _create_subcontracted_buy_rule(self, operation_type_rec, location):
-        """Creating Route Rule for Subcontracting starts here"""
-        rule = self.partner_buy_rule_id
-        if not rule:
-            buy_route = self.env.ref(
-                "purchase_stock.route_warehouse0_buy", raise_if_not_found=False
-            )
-            rule = self.env["stock.rule"].create(
-                {
-                    "name": self._compose_entity_name(),
-                    "action": "buy",
-                    "picking_type_id": operation_type_rec.id,
-                    "location_id": location.id,
-                    "route_id": buy_route.id,
-                }
-            )
-            self.partner_buy_rule_id = rule
-        return rule
-
-    def _create_subcontracted_resupply_rule(self, location):
-        """# Creating Route Rule for Subcontracting resupply on order starts here"""
-        resupply_on_order_route = self.env.ref(
-            "mrp_subcontracting.route_resupply_subcontractor_mto",
-            raise_if_not_found=False,
+        location_id = self._get_location_id_for_record(vals)
+        if "partner_picking_type_id" in vals:
+            return vals.get("partner_picking_type_id"), location_id
+        if self.partner_picking_type_id:
+            return self.partner_picking_type_id.id, location_id
```

Comment:
by this method name: the method should create a new record but you also return some values (not related with creation), maybe we can split this method? (the original method has the correct name and logic of the method, after changes the consistency of the name-logic is broken)